src/config/types.ts

```diff
@@ -4,8 +4,12 @@ export interface Dictionary<T> {
   [key: string]: T | undefined
 }
 
+// App
+export type AppMode = 'mark' | 'print' | 'mark+print'
```

Comment:
I might have done this as a sort of configuration object. That way you don't have to enumerate the mode names everywhere and encompassing a feature is easier, i.e. you don't have to say `appMode === 'print' || appMode === 'mark+print'`, you can just say `appMode.isVxPrint`. This is more useful when you have more than three configurations, but I think it still has merit here. I tried it locally and found I couldn't entirely get rid of the mode names due to serialization. Here's the diff:

```diff
diff --git a/src/AppRoot.tsx b/src/AppRoot.tsx
index 0d2f369..af9ac42 100644
--- a/src/AppRoot.tsx
+++ b/src/AppRoot.tsx
@@ -20,7 +20,6 @@ import {
   Contests,
   Election,
   ElectionDefaults,
-  InputEvent,
   MachineIdAPI,
   OptionalElection,
   OptionalVote,
@@ -28,6 +27,8 @@ import {
   UserSettings,
   VoterCardData,
   VotesDict,
+  VxMarkOnly,
+  getAppMode,
 } from './config/types'
 
 import utcTimestamp from './utils/utcTimestamp'
@@ -105,7 +106,7 @@ const initialUserState: UserState = {
 const initialState: State = {
   ...initialUserState,
   ...initialCardPresentState,
-  appMode: 'mark',
+  appMode: VxMarkOnly,
   ballotsPrintedCount: 0,
   election: undefined,
   isLiveMode: false,
@@ -447,9 +448,21 @@ class AppRoot extends React.Component<RouteComponentProps, State> {
     )
   }
 
-  public getStoredState = () => {
-    const storedState = window.localStorage.getItem(stateStorageKey)
-    return storedState ? JSON.parse(storedState) : {}
+  public getStoredState = (): Partial<State> => {
+    const storedStateJSON = window.localStorage.getItem(stateStorageKey)
+    const storedState: Partial<State> = storedStateJSON
+      ? JSON.parse(storedStateJSON)
+      : {}
+
+    if (storedState.appMode) {
+      try {
+        storedState.appMode = getAppMode(storedState.appMode.name)
+      } catch {
+        delete storedState.appMode
+      }
+    }
+
+    return storedState
   }
 
   public updateVote = (contestId: string, vote: OptionalVote) => {
@@ -520,9 +533,7 @@ class AppRoot extends React.Component<RouteComponentProps, State> {
     document.documentElement.style.fontSize = `${GLOBALS.FONT_SIZES[textSize]}px`
   }
 
-  public setAppMode = (event: InputEvent) => {
-    const currentTarget = event.currentTarget as HTMLInputElement
-    const appMode = currentTarget.dataset.appMode as AppMode
+  public setAppMode = (appMode: AppMode) => {
     this.setState({ appMode }, this.setStoredState)
   }
 
diff --git a/src/config/types.ts b/src/config/types.ts
index d6bd5a2..2286569 100644
--- a/src/config/types.ts
+++ b/src/config/types.ts
@@ -5,7 +5,35 @@ export interface Dictionary<T> {
 }
 
 // App
-export type AppMode = 'mark' | 'print' | 'mark+print'
+export interface AppMode {
+  readonly name: string
+  readonly isVxPrint?: boolean
+  readonly isVxMark?: boolean
+}
+
+export const VxPrintOnly: AppMode = { name: 'print', isVxPrint: true }
+export const VxMarkOnly: AppMode = { name: 'mark', isVxMark: true }
+export const VxMarkPlusVxPrint: AppMode = {
+  name: 'mark+print',
+  isVxPrint: true,
+  isVxMark: true,
+}
+
+export function getAppMode(name: string): AppMode {
+  switch (name) {
+    case VxPrintOnly.name:
+      return VxPrintOnly
+
+    case VxMarkOnly.name:
+      return VxMarkOnly
+
+    case VxMarkPlusVxPrint.name:
+      return VxMarkPlusVxPrint
+
+    default:
+      throw new Error(`unknown app mode: ${name}`)
+  }
+}
 
 // Events
 export type InputEvent = React.FormEvent<EventTarget>
diff --git a/src/contexts/ballotContext.ts b/src/contexts/ballotContext.ts
index 2041a4b..c738c64 100644
--- a/src/contexts/ballotContext.ts
+++ b/src/contexts/ballotContext.ts
@@ -1,11 +1,15 @@
 import { createContext } from 'react'
 import * as GLOBALS from '../config/globals'
 
-import { BallotContextInterface, TextSizeSetting } from '../config/types'
+import {
+  BallotContextInterface,
+  TextSizeSetting,
+  VxMarkOnly,
+} from '../config/types'
 
 const ballot: BallotContextInterface = {
   activateBallot: () => undefined,
-  appMode: 'mark',
+  appMode: VxMarkOnly,
   ballotStyleId: '',
   contests: [],
   election: undefined,
diff --git a/src/pages/ClerkScreen.test.tsx b/src/pages/ClerkScreen.test.tsx
index 8b5d4ec..a40f893 100644
--- a/src/pages/ClerkScreen.test.tsx
+++ b/src/pages/ClerkScreen.test.tsx
@@ -9,6 +9,7 @@ import { election } from '../../test/helpers/election'
 import { noCard, adminCard, advanceTimers } from '../../test/helpers/smartcards'
 
 import ClerkScreen from './ClerkScreen'
+import { VxMarkOnly } from '../config/types'
 
 jest.useFakeTimers()
 
@@ -18,7 +19,7 @@ fetchMock.get('/card/read', () => JSON.stringify(currentCard))
 it('renders ClerkScreen', async () => {
   const { getByText } = render(
     <ClerkScreen
-      appMode="mark"
+      appMode={VxMarkOnly}
       ballotsPrintedCount={0}
       election={election}
       fetchElection={jest.fn()}
diff --git a/src/pages/ClerkScreen.tsx b/src/pages/ClerkScreen.tsx
index af6da1a..74e053a 100644
--- a/src/pages/ClerkScreen.tsx
+++ b/src/pages/ClerkScreen.tsx
@@ -2,9 +2,11 @@ import React, { useState } from 'react'
 
 import {
   AppMode,
-  InputEventFunction,
   OptionalElection,
   VoidFunction,
+  VxMarkOnly,
+  VxPrintOnly,
+  VxMarkPlusVxPrint,
 } from '../config/types'
 
 import TestBallotDeckScreen from './TestBallotDeckScreen'
@@ -23,7 +25,7 @@ interface Props {
   isLiveMode: boolean
   fetchElection: VoidFunction
   isFetchingElection: boolean
-  setAppMode: InputEventFunction
+  setAppMode: (appMode: AppMode) => void
   toggleLiveMode: VoidFunction
   unconfigure: VoidFunction
 }
@@ -61,10 +63,7 @@ const ClerkScreen = ({
       />
     )
   }
-  const isVxMark = appMode === 'mark'
-  // const isVxPrint = appMode === 'print'
-  // const isVxMarkAndPrint = appMode === 'mark+print'
-  const isTestDecksAvailable = isLiveMode || (!isLiveMode && isVxMark)
+  const isTestDecksAvailable = isLiveMode || (!isLiveMode && appMode.isVxMark)
   return (
     <React.Fragment>
       <Main>
@@ -77,26 +76,23 @@ const ClerkScreen = ({
                 <p>This device can operate as multiple apps.</p>
                 <SegmentedButton>
                   <Button
-                    onPress={setAppMode}
-                    primary={appMode === 'mark'}
-                    disabled={appMode === 'mark'}
-                    data-app-mode="mark"
+                    onPress={() => setAppMode(VxMarkOnly)}
+                    primary={appMode === VxMarkOnly}
+                    disabled={appMode === VxMarkOnly}
                   >
                     VxMark
                   </Button>
                   <Button
-                    onPress={setAppMode}
-                    primary={appMode === 'print'}
-                    disabled={appMode === 'print'}
-                    data-app-mode="print"
+                    onPress={() => setAppMode(VxPrintOnly)}
+                    primary={appMode === VxPrintOnly}
+                    disabled={appMode === VxPrintOnly}
                   >
                     VxPrint
                   </Button>
                   <Button
-                    onPress={setAppMode}
-                    primary={appMode === 'mark+print'}
-                    disabled={appMode === 'mark+print'}
-                    data-app-mode="mark+print"
+                    onPress={() => setAppMode(VxMarkPlusVxPrint)}
+                    primary={appMode === VxMarkPlusVxPrint}
+                    disabled={appMode === VxMarkPlusVxPrint}
                   >
                     VxMark+Print
                   </Button>
@@ -133,18 +129,18 @@ const ClerkScreen = ({
                       (Available in testing mode)
                     </Text>
                   )}
-                  {!isLiveMode && isVxMark && (
+                  {!isLiveMode && !appMode.isVxPrint && (
                     <Text as="small" muted>
                       (Available with VxPrint or VxMark+Print)
                     </Text>
                   )}
                 </p>
-                <Text as="h1" muted={isVxMark}>
+                <Text as="h1" muted={!appMode.isVxPrint}>
                   Stats
                 </Text>
-                <Text muted={isVxMark}>
+                <Text muted={!appMode.isVxPrint}>
                   Printed Ballots: <strong>{ballotsPrintedCount}</strong>{' '}
-                  {isVxMark && (
+                  {!appMode.isVxPrint && (
                     <small>(Available with VxPrint or VxMark+Print)</small>
                   )}
                 </Text>
diff --git a/src/pages/PollWorkerScreen.tsx b/src/pages/PollWorkerScreen.tsx
index 63a4c7d..2ba2731 100644
--- a/src/pages/PollWorkerScreen.tsx
+++ b/src/pages/PollWorkerScreen.tsx
@@ -93,13 +93,11 @@ const PollWorkerScreen = ({
   const showModal = () => setIsModalOpen(true)
   const hideModal = () => setIsModalOpen(false)
 
-  const isVxPrint = appMode === 'print'
-  const isVxMarkAndPrint = appMode === 'mark+print'
   const currentDateTime = new Date().toLocaleString()
   const numReports = 3
 
   const togglePollsOpen = () => {
-    if (isVxPrint || isVxMarkAndPrint) {
+    if (appMode.isVxPrint) {
       showModal()
     } else {
       appTogglePollsOpen()
diff --git a/src/pages/ReviewPage.tsx b/src/pages/ReviewPage.tsx
index cd31974..264a8f4 100644
--- a/src/pages/ReviewPage.tsx
+++ b/src/pages/ReviewPage.tsx
@@ -360,7 +360,7 @@ class ReviewPage extends React.Component<RouteComponentProps, State> {
           </ContentFooter>
         </Main>
         <ButtonBar>
-          <LinkButton to={appMode === 'mark' ? '/remove' : '/print'} id="next">
+          <LinkButton to={!appMode.isVxPrint ? '/remove' : '/print'} id="next">
             Next
           </LinkButton>
           <div />
diff --git a/src/utils/utcTimestamp.ts b/src/utils/utcTimestamp.ts
index e818f54..03dc41c 100644
--- a/src/utils/utcTimestamp.ts
+++ b/src/utils/utcTimestamp.ts
@@ -1,3 +1,5 @@
-// returns current UTC unix timestamp (epoch) in seconds
+/**
+ * Returns current UTC unix timestamp (epoch) in seconds
+ */
 const utcTimestamp = (): number => Math.round(Date.now() / 1000)
 export default utcTimestamp
diff --git a/test/testUtils.tsx b/test/testUtils.tsx
index 0481290..48b69a8 100644
--- a/test/testUtils.tsx
+++ b/test/testUtils.tsx
@@ -11,10 +11,10 @@ import * as GLOBALS from '../src/config/globals'
 import electionSampleNoSeal from '../src/data/electionSampleNoSeal.json'
 
 import {
-  AppMode,
   Contests,
   Election,
   TextSizeSetting,
+  VxMarkOnly,
 } from '../src/config/types'
 
 import { mergeWithDefaults } from '../src/AppRoot'
@@ -25,7 +25,7 @@ export function render(
   {
     route = '/',
     activateBallot = jest.fn(),
-    appMode = 'mark' as AppMode,
+    appMode = VxMarkOnly,
     ballotStyleId = '',
     contests = electionSampleNoSeal.contests as Contests,
     markVoterCardUsed = jest.fn(),
```

Reply:
Thanks for your thoughts on this. I went back and forth on a couple ways to implement this. Happy to use this solution. Do you have this stashed somewhere which would make it easier to apply these changes?

Reply:
Yep, here's the commit: https://github.com/votingworks/bmd/commit/def1d6071dbd2646db38707eb79dcb46e4bbb65f